File: create-icons.py
#!/usr/bin/env python3
"""
Simple icon generator for Todo Manager PWA
Creates 192x192 and 512x512 PNG icons
"""

from PIL import Image, ImageDraw
import sys

def create_icon(size):
    """Create an icon of the specified size"""
    # Create image with blue background
    img = Image.new('RGB', (size, size), color='#3b82f6')
    draw = ImageDraw.Draw(img)

    # Scale factor
    scale = size / 512

    # Background with rounded corners (using ellipses to approximate)
    # We'll draw a solid rounded rectangle by using the image as is

    # Decorative circle
    circle_radius = int(80 * scale)
    circle_center = (size // 2, int(size * 0.39))
    draw.ellipse(
        [circle_center[0] - circle_radius, circle_center[1] - circle_radius,
         circle_center[0] + circle_radius, circle_center[1] + circle_radius],
        fill=(255, 255, 255, 76)  # Semi-transparent white
    )

    # Main checklist elements
    base_x = int(size * 0.3125)
    base_y = int(size * 0.234)

    # Function to draw checkbox
    def draw_checkbox(x, y, checked=True):
        opacity = 255 if checked else 127
        radius = int(16 * scale)
        center_x = int(x + 20 * scale)
        center_y = int(y + 20 * scale)

        # Circle
        draw.ellipse(
            [center_x - radius, center_y - radius,
             center_x + radius, center_y + radius],
            fill=(255, 255, 255, opacity)
        )

        # Checkmark
        if checked:
            line_width = max(2, int(3 * scale))
            # Simplified checkmark as lines
            draw.line(
                [(int(x + 14 * scale), int(y + 20 * scale)),
                 (int(x + 18 * scale), int(y + 24 * scale))],
                fill='#3b82f6', width=line_width
            )
            draw.line(
                [(int(x + 18 * scale), int(y + 24 * scale)),
                 (int(x + 26 * scale), int(y + 14 * scale))],
                fill='#3b82f6', width=line_width
            )

    # Function to draw line
    def draw_line(x, y, width, opacity=255):
        height = int(16 * scale)
        draw.rectangle(
            [x, y, x + width, y + height],
            fill=(255, 255, 255, opacity)
        )

    # Checkbox 1 (checked)
    draw_checkbox(base_x, base_y, True)
    draw_line(int(base_x + 50 * scale), int(base_y + 12 * scale), int(120 * scale))

    # Checkbox 2 (checked)
    draw_checkbox(base_x, int(base_y + 60 * scale), True)
    draw_line(int(base_x + 50 * scale), int(base_y + 72 * scale), int(120 * scale))

    # Checkbox 3 (unchecked)
    draw_checkbox(base_x, int(base_y + 120 * scale), False)
    draw_line(int(base_x + 50 * scale), int(base_y + 132 * scale), int(120 * scale), 127)

    # Bottom accent
    accent_x = int(size * 0.25)
    accent_y = int(size * 0.742)
    accent_width = int(size * 0.5)
    accent_height = int(8 * scale)
    draw.rectangle(
        [accent_x, accent_y, accent_x + accent_width, accent_y + accent_height],
        fill=(255, 255, 255, 76)
    )

    return img

def main():
    try:
        print('Generating icons...')

        # Generate 192x192 icon
        icon_192 = create_icon(192)
        icon_192.save('icon-192.png', 'PNG')
        print('✓ icon-192.png created')

        # Generate 512x512 icon
        icon_512 = create_icon(512)
        icon_512.save('icon-512.png', 'PNG')
        print('✓ icon-512.png created')

        print('\nIcons generated successfully!')
        return 0

    except ImportError:
        print('Error: Pillow (PIL) package not found')
        print('\nPlease install Pillow:')
        print('  pip install Pillow')
        print('\nOr use generate-icons.html in your browser to create icons manually.')
        return 1
    except Exception as e:
        print(f'Error generating icons: {e}')
        return 1

if __name__ == '__main__':
    sys.exit(main())

File: create-icons.js
#!/usr/bin/env node

/**
 * Simple icon generator for Todo Manager PWA
 * Creates 192x192 and 512x512 PNG icons
 */

const fs = require('fs');
const { createCanvas } = require('canvas');

function roundRect(ctx, x, y, width, height, radius) {
    ctx.beginPath();
    ctx.moveTo(x + radius, y);
    ctx.lineTo(x + width - radius, y);
    ctx.quadraticCurveTo(x + width, y, x + width, y + radius);
    ctx.lineTo(x + width, y + height - radius);
    ctx.quadraticCurveTo(x + width, y + height, x + width - radius, y + height);
    ctx.lineTo(x + radius, y + height);
    ctx.quadraticCurveTo(x, y + height, x, y + height - radius);
    ctx.lineTo(x, y + radius);
    ctx.quadraticCurveTo(x, y, x + radius, y);
    ctx.closePath();
}

function drawCheckbox(ctx, x, y, scale, checked) {
    const opacity = checked ? 1 : 0.5;

    // Circle
    ctx.fillStyle = `rgba(255, 255, 255, ${opacity})`;
    ctx.beginPath();
    ctx.arc(x + 20 * scale, y + 20 * scale, 16 * scale, 0, Math.PI * 2);
    ctx.fill();

    // Checkmark
    if (checked) {
        ctx.strokeStyle = '#3b82f6';
        ctx.lineWidth = 3 * scale;
        ctx.lineCap = 'round';
        ctx.beginPath();
        ctx.moveTo(x + 14 * scale, y + 20 * scale);
        ctx.lineTo(x + 18 * scale, y + 24 * scale);
        ctx.lineTo(x + 26 * scale, y + 14 * scale);
        ctx.stroke();
    }
}

function drawLine(ctx, x, y, width, scale, opacity = 1) {
    ctx.fillStyle = `rgba(255, 255, 255, ${opacity})`;
    roundRect(ctx, x, y, width, 16 * scale, 8 * scale);
    ctx.fill();
}

function generateIcon(size) {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');

    // Background with rounded corners
    const cornerRadius = size / 8;
    ctx.fillStyle = '#3b82f6';
    roundRect(ctx, 0, 0, size, size, cornerRadius);
    ctx.fill();

    // Scale factor
    const scale = size / 512;

    // Checkmark circle (decorative)
    ctx.fillStyle = 'rgba(255, 255, 255, 0.3)';
    ctx.beginPath();
    ctx.arc(size / 2, size * 0.39, 80 * scale, 0, Math.PI * 2);
    ctx.fill();

    // Main checklist
    const baseX = size * 0.3125;
    const baseY = size * 0.234;

    // Checkmark 1
    drawCheckbox(ctx, baseX, baseY, scale, true);
    drawLine(ctx, baseX + 50 * scale, baseY + 12 * scale, 120 * scale, scale);

    // Checkmark 2
    drawCheckbox(ctx, baseX, baseY + 60 * scale, scale, true);
    drawLine(ctx, baseX + 50 * scale, baseY + 72 * scale, 120 * scale, scale);

    // Checkmark 3 (unchecked)
    drawCheckbox(ctx, baseX, baseY + 120 * scale, scale, false);
    drawLine(ctx, baseX + 50 * scale, baseY + 132 * scale, 120 * scale, scale, 0.5);

    // Bottom accent
    ctx.fillStyle = 'rgba(255, 255, 255, 0.3)';
    roundRect(ctx, size * 0.25, size * 0.742, size * 0.5, 8 * scale, 4 * scale);
    ctx.fill();

    return canvas;
}

// Generate icons
try {
    console.log('Generating icons...');

    const icon192 = generateIcon(192);
    const icon512 = generateIcon(512);

    fs.writeFileSync('icon-192.png', icon192.toBuffer('image/png'));
    fs.writeFileSync('icon-512.png', icon512.toBuffer('image/png'));

    console.log('✓ icon-192.png created');
    console.log('✓ icon-512.png created');
    console.log('\nIcons generated successfully!');
} catch (error) {
    console.error('Error generating icons:', error.message);
    console.log('\nPlease install canvas package:');
    console.log('  npm install canvas');
    console.log('\nOr use generate-icons.html in your browser to create icons manually.');
    process.exit(1);
}
